From 4d7bbe40f336ea88b7f8828c454e372743e443fe Mon Sep 17 00:00:00 2001
From: Dan McGee <dan@archlinux.org>
Date: Sun, 8 Dec 2013 14:33:45 -0600
Subject: [PATCH] Fix memory leak in stdout journal streams

Just as 'identifier' is strdup-ed and freed, we need to do the same for
unit_id.

(cherry picked from commit f92ae4968f070ef0ada61ba7cd585794fac404dd)
(cherry picked from commit a345f4c7f5868d27f3c23515055ae6fa0565689d)
---
 src/journal/journald-stream.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/journal/journald-stream.c b/src/journal/journald-stream.c
index ad5df4c81d..ebe433dae6 100644
--- a/src/journal/journald-stream.c
+++ b/src/journal/journald-stream.c
@@ -339,6 +339,7 @@ void stdout_stream_free(StdoutStream *s) {
 #endif
 
         free(s->identifier);
+        free(s->unit_id);
         free(s);
 }
 
